genie-web/src/main/java/com/netflix/genie/web/configs/JobConfig.java

```diff
@@ -192,4 +197,28 @@ public WorkflowTask jobKickoffTask(
             registry
         );
     }
+
+    /**
+     * Create a {@link ProcessChecker.Factory} suitable for UNIX systems.
+     *
+     * @param executor       The executor where checks are executed
+     * @param jobsProperties The jobs properties
+     * @return a {@link ProcessChecker.Factory}
+     */
+    @Bean
+    @Autowired
+    @ConditionalOnMissingBean(ProcessChecker.Factory.class)
+    public ProcessChecker.Factory processCheckerFactory(
+        final Executor executor,
+        final JobsProperties jobsProperties
+    ) {
+        if (SystemUtils.IS_OS_UNIX) {
```

Comment:
Per my earlier comments now that I'm down here I kind of see what you're trying to do which is to fail the app coming up if it isn't supported. I still think this could be done cleaner in a centralized factory just in the constructor of said factory if the OS isn't supported.

Reply:
My thought is that it is better to fail early (during app startup) than fail in weird ways later (app is up and accepted jobs, but now we find out we don't know how to check on them)